components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -242,32 +238,26 @@ private Optional<org.wso2.carbon.user.core.common.User> getFederatedUserFromResi
         }
     }
 
-    /**
-     * Update the token binding reference with the new token id.
-     *
-     * @param tokenId Token id.
-     * @param tokenBindingRef Token binding reference.
-     * @throws IdentityOAuth2Exception If an error occurs while updating the token binding reference.
-     */
-    private void updateTokenBindingRef(String tokenId, String tokenBindingRef)
+    private void checkOrganizationIsAllowedToSwitch(String tokenIssuedOrgId, String tokenRequestedOrgId)
             throws IdentityOAuth2Exception {
 
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Updating token binding reference for token id: " + tokenId);
-        }
-        String sql = "UPDATE IDN_OAUTH2_ACCESS_TOKEN SET TOKEN_BINDING_REF=? WHERE TOKEN_ID=?";
-        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
-            try (PreparedStatement prepStmt = connection.prepareStatement(sql)) {
-                prepStmt.setString(1, tokenBindingRef);
-                prepStmt.setString(2, tokenId);
-                prepStmt.executeUpdate();
-                IdentityDatabaseUtil.commitTransaction(connection);
-            } catch (SQLException e) {
-                IdentityDatabaseUtil.rollbackTransaction(connection);
-                throw new IdentityOAuth2Exception("Error while updating the access token.", e);
+        if (StringUtils.equals(tokenIssuedOrgId, tokenRequestedOrgId)) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Provided token was already issued for the requested organization: " + tokenRequestedOrgId);
             }
-        } catch (SQLException e) {
-            throw new IdentityOAuth2Exception("Error while updating Access Token with ID: " + tokenId, e);
+            throw new IdentityOAuth2ClientException("Provided token was already issued for the requested tenant domain.");
+        }
+        try {
+            OrganizationSwitchGrantDataHolder.getInstance().getOrganizationManager()
```

Comment:
check the relative depth between orgs

Reply:
Fixed by - https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/14/commits/c5813e8893b0c0f0e0cdad0d6903ebbcc53ec4a1

The relative depth approach was not correct. Hence use the algorithm of finding the root organization and compare whether both are same.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/14/commits/629308eecec3b72a25f0934e29d59183fb4d40c0